### PR TITLE
[SPARK-6957] [SPARK-6958] [SQL] improve API compatibility to pandas

### PR DESCRIPTION
**Pull request by davies**
```
select(['cola', 'colb'])

groupby(['colA', 'colB'])
groupby([df.colA, df.colB])

df.sort('A', ascending=True)
df.sort(['A', 'B'], ascending=True)
df.sort(['A', 'B'], ascending=[1, 0])
```

cc @rxin

### CODE REVIEW

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -499,16 +502,37 @@ def sort(self, *cols):
...
+        else:
+            raise TypeError("ascending can only be bool or list, but got %s" % type(ascending))
+
+        jdf = self._jdf.sort(self._jseq(jcols))
         return DataFrame(jdf, self.sql_ctx)
 
     orderBy = sort
 
+    def _jseq(self, cols, converter=None):
+        return _to_seq(self.sql_ctx._sc, cols, converter)
```
Comment (rxin) [inline python/pyspark/sql/dataframe.py]: can we add some docstring here to explain what this does?

File: python/pyspark/sql/dataframe.py [hunk 2]
```diff
@@ -499,16 +502,37 @@ def sort(self, *cols):
...
+
+    def _jcols(self, *cols):
```
Comment (rxin) [inline python/pyspark/sql/dataframe.py]: this function too docstring

File: python/pyspark/sql/dataframe.py [hunk 3]
```diff
@@ -999,6 +1017,13 @@ def _to_java_column(col):
     return jcol
 
 
+def _to_seq(sc, cols, converter=None):
```
Comment (rxin) [inline python/pyspark/sql/dataframe.py]: docstring for this as well

File: python/pyspark/sql/dataframe.py [hunk 4]
```diff
@@ -485,13 +485,16 @@ def join(self, other, joinExprs=None, joinType=None):
         return DataFrame(jdf, self.sql_ctx)
 
     @ignore_unicode_prefix
-    def sort(self, *cols):
+    def sort(self, *cols, **kwargs):
         """Returns a new :class:`DataFrame` sorted by the specified column(s).
 
-        :param cols: list of :class:`Column` to sort by.
+        :param cols: list of :class:`Column` or column names to sort by.
+        :param ascending: sort by ascending order or not (default: True)
```
Comment (rxin) [inline python/pyspark/sql/dataframe.py]: we should explain what options are available for this, e.g. 'True', and list of ...

**Comment by rxin** [comment]
Looks good to me, except some comments on documentation. Thanks for doing this!

File: python/pyspark/sql/dataframe.py (hunk 3 above)
Comment (davies) [inline python/pyspark/sql/dataframe.py]: done

**Comment by rxin** [comment]
Thanks. I'm going to merge this and fix some minor doc issues.

**Pull request closed without merging**